pkg/apiserver/api_installer.go

```diff
@@ -148,6 +171,19 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage RESTStorage
 
 		itemPath := path + "/{name}"
 		nameParams := append(params, nameParam)
+		namespaceFn = func(req *restful.Request) (namespace string, err error) {
+			return
```

Comment:
> Naked returns are okay if the function is a handful of lines.

I think these functions can be moved to static functions, they don't need to be closures.  Will do that in a follow up.

----- Original Message -----
> > @@ -148,6 +171,19 @@ func (a *APIInstaller) registerResourceHandlers(path
> > string, storage RESTStorage
> >  
> >  		itemPath := path + "/{name}"
> >  		nameParams := append(params, nameParam)
> > +		namespaceFn = func(req *restful.Request) (namespace string, err error) {
> > +			return
> 
> Yuck. Naked returns in the middle of a large function are incredibly hard to
> read. See
> https://code.google.com/p/go-wiki/wiki/CodeReviewComments#Named_Result_Parameters
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4263/files#r24597352